backend/protocol_rpc/message_handler/base.py

```diff
@@ -1,65 +1,66 @@
-import traceback
-import random
-import string
 import os
 import json
-from functools import wraps, partial
+from functools import wraps
 from logging.config import dictConfig
-from flask import Flask
+import traceback
+from loguru import logger
+import sys
+
+from backend.protocol_rpc.message_handler.types import LogEvent
 from flask_socketio import SocketIO
 
 from backend.protocol_rpc.configuration import GlobalConfiguration
-from backend.protocol_rpc.message_handler.types import FormattedResponse
-from backend.protocol_rpc.types import EndpointResult, EndpointResultStatus
+from backend.protocol_rpc.message_handler.types import EventScope, EventType, LogEvent
 
 MAX_LOG_MESSAGE_LENGTH = 3000
 
 
 class MessageHandler:
-
-    status_mappings = {
-        "debug": "info",
-        "info": "info",
-        "success": "info",
-        "error": "error",
-    }
-
-    def __init__(self, app: Flask, socketio: SocketIO, config: GlobalConfiguration):
-        self.app = app
+    def __init__(self, socketio: SocketIO, config: GlobalConfiguration):
         self.socketio = socketio
         self.config = config
         setup_logging_config()
 
     def log_endpoint_info(self, func):
         return log_endpoint_info_wrapper(self, self.config)(func)
 
-    def socket_emit(self, message):
-        self.socketio.emit("status_update", {"message": message})
-
-    def log_message(self, function_name, result: EndpointResult):
-        logging_status = self.status_mappings[result.status.value]
-        if hasattr(self.app.logger, logging_status):
-            log_method = getattr(self.app.logger, logging_status)
-            result_string = str(result)
-            function_result = (
-                (result_string[:MAX_LOG_MESSAGE_LENGTH] + "...")
-                if result_string is not None
-                and len(result_string) > MAX_LOG_MESSAGE_LENGTH
-                else result_string
-            )
-            log_method(function_name + ": " + function_result)
-        else:
-            raise Exception(f"Logger does not have the method {result.status}")
+    def socket_emit(self, log_event: LogEvent):
```

Comment:
nit: prepending with `_` is a way to mark internal methods in Python. It might be a good idea to do that so that we don't accidentally manually call this method for example

Reply:
ok, I shall do the same for both methods then

Reply:
![Uploading image.png…]()

Reply:
<img width="422" alt="image" src="https://github.com/user-attachments/assets/c94c4129-b94c-4e06-a959-8e514f18e3c8">